unlock-app/src/components/interface/locks/CheckoutUrl/elements/DynamicForm.tsx

```diff
@@ -0,0 +1,218 @@
+import { Button, Input, Select } from '@unlock-protocol/ui'
+import { z } from 'zod'
+import zodToJsonSchema from 'zod-to-json-schema'
+import { FormProvider, useForm, useFormContext } from 'react-hook-form'
+
+interface DynamicFormProps {
+  name: string
+  schema: z.Schema
+  title?: string
+  description?: Record<string, string>
+  onChange: (fields: any) => void
+  onSubmit?: (fields: any) => void
+  submitLabel?: string
+  defaultValues?: any
+  showSubmit?: boolean
+}
+
+interface ComponentByTypeMapProps {
+  [type: string]: any
+}
+
+interface FieldProps {
+  required?: boolean
+  label?: string
+  type: string
+  name: string
+  props: Record<string, any>
+}
+
+export const ConnectForm = ({ children }: any) => {
+  const methods = useFormContext()
+
+  return children({ ...methods })
+}
+
+const TextInput = ({ props, type, ...rest }: FieldProps) => {
+  const { enum: enumList = [] } = props ?? {}
+  const hasOptions = enumList?.length
+
+  if (!hasOptions) {
+    return (
+      <ConnectForm>
+        {({ register }: any) => <Input {...register(rest.name)} {...rest} />}
+      </ConnectForm>
+    )
+  }
+
+  const options = enumList.map((enumItem: string) => ({
+    label: enumItem,
+    value: enumItem,
+  }))
+
+  return <Select options={options} {...rest} />
+}
+
+const BooleanInput = ({ props, name, label, ...rest }: any) => {
+  return (
+    <div className="block gap-3 py-2 ">
+      <ConnectForm>
+        {({ register }: any) => (
+          <>
+            <div className="flex flex-col gap-1">
+              <div className="flex items-center gap-3">
+                <input
+                  type="checkbox"
+                  id={name}
+                  name={name}
+                  className="outline-none focus:outline-none hover:outline-none"
+                  {...register(name)}
+                />
+                <label htmlFor={name}>{label}</label>
+              </div>
+              {rest?.description && (
+                <span className="text-xs text-gray-600 ">
+                  {rest?.description}
+                </span>
+              )}
+            </div>
+          </>
+        )}
+      </ConnectForm>
+    </div>
+  )
+}
+
+const ComponentByTypeMap: ComponentByTypeMapProps = {
+  string: TextInput,
+  integer: TextInput,
+  boolean: BooleanInput,
+}
+
+const TypeMap: Record<string, string> = {
+  string: 'text',
+  integer: 'number',
+}
+
+const camelCaseToText = (text: string) => {
```

Comment:
Hum, unfortunately this is not the right approach here. We need to have a mapping of `label` like we have a mapping of `description`.